util/stopper_test.go

```diff
@@ -237,3 +236,52 @@ func TestStopperClosers(t *testing.T) {
 		t.Errorf("expected true & true; got %t & %t", tc1, tc2)
 	}
 }
+
+func TestStopperNumTasks(t *testing.T) {
+	s := NewStopper()
+
+	var tasks []PendingTask
+	for i := 0; i < 3; i++ {
+		tasks = append(tasks, s.StartTask())
+		if numTypes, numTasks := len(s.tasks), s.NumTasks(); numTypes != 1 || numTasks != i+1 {
+			t.Errorf("stopper should have %d running tasks, got %d / %+v", i+1, s.NumTasks(), s.tasks)
+		}
+		m := s.RunningTasks()
+		if len(m) != 1 {
+			t.Fatalf("expected exactly one task map entry: %+v", m)
+		}
+		for _, v := range m {
+			if v != i+1 {
+				t.Fatalf("expected %d tasks, got %d", i+1, v)
+			}
+		}
+	}
+
+	var i int
+	for _, task := range tasks {
+		i++
+		if !task.Ok() {
+			t.Fatalf("task should have been allowed")
+		}
+		task.Done()
+		if numTypes, numTasks, expNum := len(s.tasks), s.NumTasks(), 3-i; numTypes != 1 || numTasks != expNum {
```

Comment:
this `3-i` is really len(tasks[i:]) or len(tasks[i+1:]) if `i` is zero indexed as i suggested above